### PR TITLE
Update index-state

### PR DESCRIPTION
**Pull request by newhoggy**
* Update index-state
* Add lower-bounds to nonempty-containers
* Do not define ToJSON/FromJSON instances of `NESet`

The instance fix addresses this error when building with new index-state:

```
src/Cardano/Prelude/Orphans.hs:27:10: error:
    Duplicate instance declarations:
      instance (Ord a, FromJSON a) => FromJSON (NESet a)
        -- Defined at src/Cardano/Prelude/Orphans.hs:27:10
      instance (FromJSON a, Ord a) => FromJSON (NESet a)
        -- Defined in ‘Data.Set.NonEmpty.Internal’
   |
27 | instance (Ord a, FromJSON a) => FromJSON (NESet a) where
   |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

src/Cardano/Prelude/Orphans.hs:30:10: error:
    Duplicate instance declarations:
      instance (Ord a, ToJSON a) => ToJSON (NESet a)
        -- Defined at src/Cardano/Prelude/Orphans.hs:30:10
      instance ToJSON a => ToJSON (NESet a)
        -- Defined in ‘Data.Set.NonEmpty.Internal’
   |
30 | instance (Ord a, ToJSON a) => ToJSON (NESet a) where
   |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

### CODE REVIEW

File: src/Cardano/Prelude/Orphans.hs [hunk 1]
```diff
@@ -1,35 +1,25 @@
+{-# LANGUAGE CPP               #-}
 {-# LANGUAGE OverloadedStrings #-}
 {-# LANGUAGE PolyKinds         #-}
 {-# LANGUAGE TypeFamilies      #-}
 
 {-# OPTIONS_GHC -fno-warn-orphans #-}
+{-# OPTIONS_GHC -fno-warn-unused-imports #-}
```
Comment (erikd) [inline src/Cardano/Prelude/Orphans.hs:7]: What imports are being warned about?

Comment (newhoggy) [reply src/Cardano/Prelude/Orphans.hs:7]: Oops.  That was a carry over from a previous commit.  I'll remove that.

Comment (newhoggy) [reply src/Cardano/Prelude/Orphans.hs]: Fixed!

File: src/Cardano/Prelude/Orphans.hs [hunk 2]
```diff
@@ -1,3 +1,4 @@
+{-# LANGUAGE CPP               #-}
```
Comment (erikd) [inline src/Cardano/Prelude/Orphans.hs:1]: And why do we need that?

Comment (newhoggy) [reply src/Cardano/Prelude/Orphans.hs]: Use some CPP previously, but realised that I didn't need to.  Removed.  Thanks!

**Review by erikd — APPROVED** [review]

**Comment by newhoggy** [comment]
bors merge

**Review by intricate — COMMENTED** [review]
About 15 hours late, but LGTM :smile:

**Pull request merged**